stripe/src/main/java/com/stripe/android/model/Address.kt

```diff
@@ -0,0 +1,176 @@
+package com.stripe.android.model
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.stripe.android.ObjectBuilder
+import com.stripe.android.model.StripeJsonUtils.optString
+import java.util.Locale
+import java.util.Objects
+import org.json.JSONObject
+
+/**
+ * Model for an owner [address](https://stripe.com/docs/api#source_object-owner-address)
+ * object in the Source api.
+ */
+class Address : StripeModel, StripeParamsModel, Parcelable {
+    val city: String?
+    val country: String?
+    val line1: String?
+    val line2: String?
+    val postalCode: String?
+    val state: String?
+
+    private constructor(
+        city: String?,
+        country: String?,
+        line1: String?,
+        line2: String?,
+        postalCode: String?,
+        state: String?
+    ) {
+        this.city = city
+        this.country = country
+        this.line1 = line1
+        this.line2 = line2
+        this.postalCode = postalCode
+        this.state = state
+    }
+
+    private constructor(addressBuilder: Builder) : this(
+        addressBuilder.city,
+        addressBuilder.country,
+        addressBuilder.line1,
+        addressBuilder.line2,
+        addressBuilder.postalCode,
+        addressBuilder.state
+    )
+
+    private constructor(parcel: Parcel) {
+        city = parcel.readString()
+        country = parcel.readString()
+        line1 = parcel.readString()
+        line2 = parcel.readString()
+        postalCode = parcel.readString()
+        state = parcel.readString()
+    }
+
+    override fun toParamMap(): Map<String, Any> {
+        return mapOf(
+            FIELD_CITY to city.orEmpty(),
+            FIELD_COUNTRY to country.orEmpty(),
+            FIELD_LINE_1 to line1.orEmpty(),
+            FIELD_LINE_2 to line2.orEmpty(),
+            FIELD_POSTAL_CODE to postalCode.orEmpty(),
+            FIELD_STATE to state.orEmpty()
+        ).filterValues { it.isNotEmpty() }
+    }
+
+    override fun equals(other: Any?): Boolean {
+        return when {
+            this === other -> true
+            other is Address -> typedEquals(other)
+            else -> false
+        }
+    }
+
+    private fun typedEquals(address: Address): Boolean {
+        return city == address.city && country == address.country && line1 == address.line1 &&
+            line2 == address.line2 && postalCode == address.postalCode && state == address.state
+    }
+
+    override fun hashCode(): Int {
+        return Objects.hash(city, country, line1, line2, postalCode, state)
+    }
+
+    override fun writeToParcel(out: Parcel, flags: Int) {
+        out.writeString(city)
+        out.writeString(country)
+        out.writeString(line1)
+        out.writeString(line2)
+        out.writeString(postalCode)
+        out.writeString(state)
+    }
+
+    override fun describeContents(): Int {
+        return 0
+    }
+
+    class Builder : ObjectBuilder<Address> {
```

Comment:
if you make this a data class you don't need a builder I guess? maybe you do if it's used from java?

Reply:
I want to make the constructor private, and Kotlin warns me about a private constructor with a data class. Is there a workaround?

Reply:
ah yeah I dont think that's possible